rtsp.go

```diff
@@ -658,11 +675,26 @@ func newRTSPCamera(ctx context.Context, _ resource.Dependencies, conf resource.C
 	}
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	reader := gostream.VideoReaderFunc(func(_ context.Context) (image.Image, func(), error) {
-		latest := rc.latestFrame.Load()
+		latest := rc.latestOutput.Load()
 		if latest == nil {
 			return nil, func() {}, errors.New("no frame yet")
 		}
-		return *latest, func() {}, nil
+
+		release := func() {}
+		// When model is mjpeg, latest frame wrapper will be nil.
+		if latest.frameWrapper != nil {
+			latest.frameWrapper.isBeingServed.Store(true)
```

Comment:
I'm pretty sure `isBeingServed` is equivalent to `isInPool`? This feels like an exercise where the user should remember to clal `isBeingServed(true)` after every `pool.get` and `isBeingServed(false)` prior to every `pool.put`. I don't think `isBeingServed`s access pattern will ever diverge from `isInPool`? But maybe I'm missing something.

Reply:
Added ref counting